### ISSUE TITLE
Capture groups in tags interfere with capture groups from your library

### ISSUE THREAD

**Issue by telotortium** [alexflint/go-restructure#7]
Modify your test example as follows (BTW, `\x60` is the backtick, since I was trying to figure out how to use it inside a regex without removing the string literal backticks):

```
type DotExpr struct {
        _    struct{} `^`
        Head string   `(\w|\x60)+`
        Tail *DotName `?`
        _    struct{} `$`
}
```

Note that `Head` now includes a capture group. The associated test will now fail:

```
$ go test
--- FAIL: TestMatchNameDotName (0.00s)
        Error Trace:    restructure_test.go:28
        Error:          Not equal: "foo" (expected)
                                != "o" (actual)

--- FAIL: TestMatchNameDotNameHeadOnly (0.00s)
        Error Trace:    restructure_test.go:40
        Error:          Not equal: "head" (expected)
                                != "d" (actual)

FAIL
exit status 1
FAIL    github.com/alexflint/go-restructure     0.003s
```

This shows that the capture groups specified in the struct tags interfere with those from the library. It can be worked around by using non-capture groups: `(?:\w|\x60)+`. You should address this: either warn the user not to use capture groups, parse the regexps in `restructure.Compile` for capture groups and return an error if any are found, or figure out how to handle them without breaking your library.

**Comment by telotortium** [comment]
Oh, and of course, add tests with fields including both capture and non-capture groups.

**Comment by alexflint** [comment]
Ah yes, a known problem I'm afraid. Need to walk the AST generated by `syntax.Compile` for terminals and renumber the capture groups.

**Comment by zackbloom** [comment]
I ran into this too.  Perhaps it would be possible for the library to error if a capturing group is included in the struct?

**Comment by alexflint** [comment]
This is resolved by #9
